File: TaskTracker_BE/TaskTracker.Api/Data/Models/Ticket.cs
using TaskTracker.Api.Data.Enums;

namespace TaskTracker.Api.Data.Models;

public class Ticket : Base
{
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public EStatus Status { get; set; }
    public EPriority Priority { get; set; }
    public DateTime? DueDate { get; set; }
}

File: TaskTraker.Web/src/app/models/Ticket.ts
export enum EStatus {
  New = 0,
  InProgress = 1,
  Done = 2,
}

export enum EPriority {
  Low = 0,
  Medium = 1,
  High = 2,
}

export interface Ticket {
  id?: number;
  title: string;
  description: string;
  status: EStatus;
  priority: EPriority;
  dueDate?: string | null;
  createdAt?: string;
}
